### PR TITLE
compat: module resolution, 'deno' conditional exports, add example

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

**Comment by bartlomieju** [comment]
CC @guybedford

**Comment by bartlomieju** [comment]
Before landing I want to add information about usage of package managers and add example of running `eslint` with compat mode.

File: npm_nodejs/compatibility_mode.md [hunk 1]
```diff
@@ -1,31 +1,108 @@
...
+  1:17  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
+  1:21  error    Unexpected empty function 'bar'           @typescript-eslint/no-empty-function
+
+✖ 5 problems (2 errors, 3 warnings)
+```
+
+> ⚠️ Notice that ESLint is run with limited set of permissions. We only give it
+> access to the read from the file system, write to current directory and access
+> environmental variables. Programs run in compatility mode are subject to
+> Deno's permission model.
```
Comment (kt3k) [inline npm_nodejs/compatibility_mode.md:54]: This is a great advantage of using deno compat mode over Node.js itself 👍

**Review by kt3k — APPROVED** [review]
LGTM

File: npm_nodejs/compatibility_mode.md (hunk 1 above)
Comment (bartlomieju) [reply npm_nodejs/compatibility_mode.md:54]: Indeed!

**Pull request merged**